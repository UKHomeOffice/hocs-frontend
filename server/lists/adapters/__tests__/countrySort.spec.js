const countrySortAdapter = require('../countrySort');

const mockLogger = {
    debug: () => { },
    info: () => { },
    warn: () => { },
    error: () => { }
};

describe('Country Sort Adapter', () => {

    it('should transform and sort country data', async () => {
        const mockData = [
            'Country B',
            'Country C',
            'Country A'
        ];

        const results = await countrySortAdapter(mockData, { logger: mockLogger });

        expect(results).toBeDefined();
        expect(results).toMatchSnapshot();
    });

    it('should sort United Kingdom to top', async () => {
        const mockData = [
            'Country B',
            'Country A',
            'United Kingdom'
        ];

        const results = await countrySortAdapter(mockData, { logger: mockLogger });

        expect(results).toBeDefined();
        expect(results).toMatchSnapshot();
    });
});