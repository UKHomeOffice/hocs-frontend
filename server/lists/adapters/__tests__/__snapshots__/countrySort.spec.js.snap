// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Country Sort Adapter should sort United Kingdom to top 1`] = `
Array [
  Object {
    "label": "United Kingdom",
    "value": "United Kingdom",
  },
  Object {
    "label": "Country A",
    "value": "Country A",
  },
  Object {
    "label": "Country B",
    "value": "Country B",
  },
]
`;

exports[`Country Sort Adapter should transform and sort country data 1`] = `
Array [
  Object {
    "label": "Country A",
    "value": "Country A",
  },
  Object {
    "label": "Country B",
    "value": "Country B",
  },
  Object {
    "label": "Country C",
    "value": "Country C",
  },
]
`;
